explain query does not work if it has a - (Hyphen) in the query name

**Describe the bug**
Trying to do a `explain` on a query which has `-` (Hyphen) in it, does not work. 

```
ksql> show queries;

 Query ID                              | Status    | Sink Name                      | Sink Kafka Topic               | Query String
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 CTAS_PN-CLICKED-COUNTER_13            | RUNNING:4 | pn-clicked-counter             | pn-clicked-counter             | MY_QUERY

ksql> explain CTAS_PN-CLICKED-COUNTER_13;
line 1:16: mismatched input '-' expecting ';'
Statement: explain CTAS_PN-CLICKED-COUNTER_13;
Caused by: line 1:16: mismatched input '-' expecting ';'
Caused by: org.antlr.v4.runtime.InputMismatchException

ksql> explain `CTAS_PN-CLICKED-COUNTER_13`;
Query with id:`CTAS_PN-CLICKED-COUNTER_13` does not exist, use SHOW QUERIES to view the full set of queries.

ksql> explain "CTAS_PN-CLICKED-COUNTER_13";
Query with id:"CTAS_PN-CLICKED-COUNTER_13" does not exist, use SHOW QUERIES to view the full set of queries.

ksql> explain 'CTAS_PN-CLICKED-COUNTER_13';
line 1:9: mismatched input ''CTAS_PN-CLICKED-COUNTER_13'' expecting {'EMIT', 'CHANGES', 'SELECT', 'ESCAPE', 'INTEGER', 'DATE', 'TIME', 'TIMESTAMP', 'INTERVAL', 'YEAR', 'MONTH', 'DAY', 'HOUR', 'MINUTE', 'SECOND', 'ZONE', 'PARTITION', 'STRUCT', 'CREATE', 'INSERT', 'DESCRIBE', 'PRINT', 'EXPLAIN', 'ANALYZE', 'TYPE', 'TYPES', 'SHOW', 'LIST', 'TABLES', 'TERMINATE', 'COLUMNS', 'COLUMN', 'PARTITIONS', 'FUNCTIONS', 'FUNCTION', 'DROP', 'ARRAY', 'MAP', 'SET', 'RESET', 'SESSION', 'UNSET', 'KEY', 'SINK', 'SOURCE', 'PRIMARY', 'IF', IDENTIFIER, DIGIT_IDENTIFIER, QUOTED_IDENTIFIER, BACKQUOTED_IDENTIFIER}
Statement: explain 'CTAS_PN-CLICKED-COUNTER_13';
Caused by: line 1:9: mismatched input ''CTAS_PN-CLICKED-COUNTER_13'' expecting
	{'EMIT', 'CHANGES', 'SELECT', 'ESCAPE', 'INTEGER', 'DATE', 'TIME', 'TIMESTAMP',
	'INTERVAL', 'YEAR', 'MONTH', 'DAY', 'HOUR', 'MINUTE', 'SECOND', 'ZONE',
	'PARTITION', 'STRUCT', 'CREATE', 'INSERT', 'DESCRIBE', 'PRINT', 'EXPLAIN',
	'ANALYZE', 'TYPE', 'TYPES', 'SHOW', 'LIST', 'TABLES', 'TERMINATE', 'COLUMNS',
	'COLUMN', 'PARTITIONS', 'FUNCTIONS', 'FUNCTION', 'DROP', 'ARRAY', 'MAP', 'SET',
	'RESET', 'SESSION', 'UNSET', 'KEY', 'SINK', 'SOURCE', 'PRIMARY', 'IF',
	IDENTIFIER, DIGIT_IDENTIFIER, QUOTED_IDENTIFIER, BACKQUOTED_IDENTIFIER}
Caused by: org.antlr.v4.runtime.InputMismatchException
```

**Expected behavior**
`explain queryId` should give the execution plan for the `queryId`

--- a/ksqldb-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java
+++ b/ksqldb-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java
@@ -158,7 +158,6 @@ import java.util.concurrent.TimeUnit;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 import org.antlr.v4.runtime.ParserRuleContext;
-import org.antlr.v4.runtime.RuleContext;
 import org.antlr.v4.runtime.Token;
 import org.antlr.v4.runtime.tree.ParseTree;
 import org.antlr.v4.runtime.tree.TerminalNode;
@@ -1175,7 +1174,7 @@ public class AstBuilder {
     public Node visitExplain(final SqlBaseParser.ExplainContext ctx) {
       final IdentifierContext queryIdentifier = ctx.identifier();
       final Optional<String> queryId = Optional.ofNullable(queryIdentifier)
-          .map(RuleContext::getText);
+          .map(ParserUtil::getIdentifierText);
 
       final Optional<Statement> statement = Optional.ofNullable(ctx.statement())
           .map(s -> (Statement) visit(s));

--- a/ksqldb-parser/src/test/java/io/confluent/ksql/parser/AstBuilderTest.java
+++ b/ksqldb-parser/src/test/java/io/confluent/ksql/parser/AstBuilderTest.java
@@ -36,6 +36,7 @@ import io.confluent.ksql.parser.KsqlParser.ParsedStatement;
 import io.confluent.ksql.parser.SqlBaseParser.SingleStatementContext;
 import io.confluent.ksql.parser.tree.AliasedRelation;
 import io.confluent.ksql.parser.tree.AllColumns;
+import io.confluent.ksql.parser.tree.Explain;
 import io.confluent.ksql.parser.tree.Join;
 import io.confluent.ksql.parser.tree.Query;
 import io.confluent.ksql.parser.tree.QueryContainer;
@@ -529,6 +530,18 @@ public class AstBuilderTest {
     // Then:
     assertThat("Should be push", result.isPullQuery(), is(false));
     assertThat(result.getResultMaterialization(), is(ResultMaterialization.CHANGES));
+  }
+
+  @Test
+  public void shouldSupportQuotedExplainStatements() {
+    // Given:
+    final SingleStatementContext stmt = givenQuery("EXPLAIN `CSAS_FOO-BAR`;");
+
+    // When:
+    final Explain explain = (Explain) builder.buildStatement(stmt);
+
+    // Then:
+    assertThat(explain.getQueryId(), is(Optional.of("CSAS_FOO-BAR")));
   }
 
   private static SingleStatementContext givenQuery(final String sql) {

--- a/ksqldb-parser/src/test/java/io/confluent/ksql/parser/SqlFormatterTest.java
+++ b/ksqldb-parser/src/test/java/io/confluent/ksql/parser/SqlFormatterTest.java
@@ -629,7 +629,7 @@ public class SqlFormatterTest {
 
     final String result = SqlFormatter.formatSql(statement);
 
-    assertThat(result, is("EXPLAIN \nfoo"));
+    assertThat(result, is("EXPLAIN \nFOO"));
   }
 
   @Test

